pkg/reconciler/common/prune.go

```diff
@@ -170,5 +301,39 @@ func checkAndDelete(k kubernetes.Interface, ctx context.Context, targetNamespace
 	}
 
 	//if there is no error it means cron is exists, but no prune in config it means delete it
-	return k.BatchV1beta1().CronJobs(targetNamespace).Delete(ctx, CronName, v1.DeleteOptions{})
+	return pruner.kc.BatchV1beta1().CronJobs(ns).Delete(pruner.context, CronName+"-"+ns, v1.DeleteOptions{})
+}
+
+func (pruner *Pruner) removedFromTektonConfig(prune v1alpha1.Prune) bool {
+	if len(prune.Resources) == 0 || prune.Schedule == "" {
+		return true
+	}
+	return false
+}
+
+func (pruner *Pruner) removedNSAnnotation() ([]string, bool) {
+	var namespaces []string
+	cronJobs, err := pruner.kc.BatchV1beta1().CronJobs("").List(pruner.context, v1.ListOptions{})
```

Comment:
let us add a label to all the cronJobs we create eg: `tektonconfig.operator.tekton.dev/pruner: true`

then this cronJob query can be improved by passing the label trough listOptions

Reply:
Sure